src/machines/authMachine.ts

```diff
@@ -280,3 +350,19 @@ async function logout() {
     credentials: 'include',
   })
 }
+
+/**
+ * To logout you need to revoke the token via the `oauth2/token/revoke` deleting the token off disk for electron
+ * will not be sufficient.
+ */
+async function logoutAllEnvironments() {
+  if (!isDesktop()) {
+    return new Error('unimplemented for web')
+  }
+  const environments = await listAllEnvironments()
```

Comment:
Should this be typed? I see `Promise<any>` here

Asking because I was about to suggest `for (const environmentName of environments)` for the next two lines